routers/api/v1/api.go

```diff
@@ -955,7 +963,7 @@ func Routes() *web.Router {
 
 		// Users (requires user scope)
 		m.Group("/users", func() {
-			m.Get("/search", reqExploreSignIn(), user.Search)
+			m.Get("/search", reqExploreSignIn(), reqUsersExploreEnabled(), user.Search)
```

Comment:
Why this limit is only added to API? We also have an endpoint in web side.
![image](https://github.com/user-attachments/assets/f05c2be6-28ce-43ae-aac7-3a2e8099ddea)

So does this setting mean disable user search? Then the name should be `DisableUsersSearch` and not in service.explore section? I can not understand.

Reply:
Hmmm. I think when admins set `DISABLE_USERS_PAGE` to true, they intend to disable other users from getting a list of all users. So I disable `/api/v1/user/search` API in this case. But `/user/search` is used to add team members and repo collaborators, if we disable user search in this case, users will not be able to use these features.

Reply:
But I can also get a list of all users from `/user/search` this endpoint.

Reply:
`/user/search`  is only used by SearchUser.ts, so maybe it could be fixed like this: make it only: drop unused parameters and make it only return the first N results.

Reply:
> But I can also get a list of all users (actor can see) from `/user/search` Then this problem becomes a security problem.

I agree with your opinion that this may be a security issue. But this PR is to add the `DISABLE_ORGANIZATIONS_PAGE` and `DISABLE_CODE_PAGE` settings and this issue is caused by the incorrect usage of `DISABLE_USERS_PAGE`. I think we could create another PR to resolve this issue.

Reply:
done in ea0277867db5ea996f4faea9018ddb5d10b1493b

Reply:
> done in [ea02778](https://github.com/go-gitea/gitea/commit/ea0277867db5ea996f4faea9018ddb5d10b1493b)

Thank you so much!

Reply:
As the title, yes, this PR is not related to `DISABLE_USERS_PAGE`.
But you touched it here, does `DISABLE_ORGANIZATIONS_PAGE` and `DISABLE_CODE_PAGE` settings related to this API? That's why this review comment disappeared.